examples/00-mapdl-examples/BodyFlux-Averaging.py

```diff
@@ -0,0 +1,324 @@
+mapdl.nsel("EXT")  # Select exterior nodes
+mapdl.d("ALL", "AZ", 0)  # Set potentials to zero (flux-parallel)
+
+###############################################################################
+# Solve the model
+# ===============
+#
+# Solve the magnetostatic analysis.
+
+mapdl.allsel("ALL")
+mapdl.solve()
+mapdl.finish()
+
+###############################################################################
+# Post-processing
+# ===============
+#
+# Open the result file and read in the last set of results
+
+mapdl.post1()
+mapdl.file("file", "rmg")
+mapdl.set("last")
+
+###############################################################################
+#
+# Printing the nodal values
+#
+
+print(mapdl.post_processing.nodal_values("b", "x"))
+
+###############################################################################
+# Create an MAPDL Power Graphics plot of the X direction Magnetic Flux
+# --------------------------------------------------------------------
+#
+# The MAPDL colors are reversed via the rgb command so that the background is
+# white with black text and element edges.
+
+mapdl.graphics("power")
+mapdl.rgb("INDEX", 100, 100, 100, 0)
+mapdl.rgb("INDEX", 80, 80, 80, 13)
+mapdl.rgb("INDEX", 60, 60, 60, 14)
+mapdl.rgb("INDEX", 0, 0, 0, 15)
+
+mapdl.edge(1, 1)
+mapdl.show("png")
+mapdl.pngr("tmod", 0)
+
+mapdl.plnsol("b", "x")
+mapdl.show("")
+
+###############################################################################
+# Obtain Grid and Scalar Data
+# ---------------------------
+#
+# First, obtain the set of unique material IDs in the model
+
+elem_mats = mapdl.mesh.material_type
+np.unique(elem_mats)
+
+###############################################################################
+# For each unique material ID the elements are selected, and so are their
+# nodes.
+# The 'grids' list is appended with the mesh information of just those
+# elements, and the scalars list appended with the nodal X direction magnetic
+# flux.
+
+grids = []
+scalars = []
+for mat in np.unique(elem_mats):
+    mapdl.esel("s", "mat", "", mat)
+    mapdl.nsle()
+    grids.append(mapdl.mesh.grid)
+    scalars.append(mapdl.post_processing.nodal_values("b", "x"))
+mapdl.allsel()
+
+###############################################################################
+# If interested print the grids list and perhaps compare to a print of
+# mapdl.mesh.grid
+
+print(grids)
+# print(mapdl.mesh.grid)
+
+
+###############################################################################
+# Color Map and Result Plot
+# -------------------------
+#
+# Some of the MAPDL contour colors did not have an exact match in the standard
+# Matplotlib color library so an attempt was made to match the color and use
+# the Hex RGBA number value.
+#
+# For each item in the grids list the grid is added to the plot and 9 contour
+# colors requested using the prior define color map and the same contour
+# legend.
+#
+# The plot is then shown and it recreates the native plot quite well.
```

Comment:
```suggestion
# Color map and result plot
# -------------------------
#
# Because some of the MAPDL contour colors do not have an exact match in the
# standard Matplotlib color library, an attempt is made to match the color and use
# the Hex RGBA number value.
#
# For each item in the grids list the grid is added to the plot and 9 contour
# colors requested using the prior define color map and the same contour
# legend.
#
# The plot is then shown and it recreates the native plot quite well.
```